### PR TITLE
Make documentation guidelines easier to discover

### PR DESCRIPTION
**Pull request by ryan-blunden**
- Added a link to the documentation guidelines from the engineering home page page index to help with discovery.
- Moves the documentation page into the engineering directory
- Added a paragraph to the documentation page explaining its documentation for the sourcegraph repo while also adding link for editing the handbook
- Added a link to editing the handbook on the handbook home page
- Added a > NOTE section to the handbook usage page to clarify that the documentation page is for adding info about using or configuring Sourcegraph (the product).

### CODE REVIEW

File: handbook/index.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ The Sourcegraph handbook describes how we (Sourcegraph teammates) work. It's pub
 ## Engineering
 
 - [Engineering](engineering/index.md)
+- [Documentation](documentation/index.md)
```
Comment (nicksnyder) [inline handbook/index.md]: Why is this under engineering and not a new top level item (as the directory would suggest)? There is a bit of ambiguity here whether that page is product documentation or handbook documentation.

```suggestion

## Documentation

- [Documentation](documentation/index.md)
```

Comment (ryan-blunden) [reply handbook/index.md]: > There is a bit of ambiguity here whether that page is product documentation or handbook documentation

@nicksnyder I agree and here are some additional changes I've made to help address this:

- Removed the documentation link on the home page
- I've pushed a new change that moves the documentation page under engineering
- I've added a paragraph to that documentation page explaining its documentation for the sourcegraph repo with a link to jump to how to edit the handbook
- Added a link to editing the handbook on the handbook home page

File: handbook/engineering/documentation.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
-# Sourcegraph documentation guidelines
+# Sourcegraph engineering documentation
```
Comment (nicksnyder) [inline handbook/engineering/documentation.md]: Let's call this product documentation. Can we rename the file also to `product_documentation.md`?

```suggestion
# Product documentation
```

File: handbook/engineering/index.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 - [Code reviews](code_reviews.md)
 - [Go style guide](go_style_guide.md)
 - [Testing](testing.md)
+- [Documentation](documentation.md)
```
Comment (nicksnyder) [inline handbook/engineering/index.md]:
```suggestion
- [Product documentation](product_documentation.md)
```

File: handbook/usage.md [hunk 4]
```diff
@@ -29,7 +29,9 @@ Join the `#handbook` Slack channel to see proposed changes.
    - In your message announcing the change, just include `[Handbook/process] change:`, the commit message subject, and the link. Don't write a separate summary of the change (if you feel the need to do so, you should have used a better commit message).
    - Ensure everyone who needs to be aware of the change is made aware. If you see people do things that indicate they aren't aware, gently remind them of the change. If you don't nudge them, other people will see them doing things the wrong/old way and start to doubt the handbook's accuracy or currency.
 1. Strongly prefer creating a pull request over filing an issue on the handbook or mentioning it on Slack, even if you aren't able to propose the full edits you'd like to make.
-1. Follow the [style guide](communication/style_guide.md) and [documentation guidelines](documentation/index.md) when making changes to the handbook.
+1. Follow the [style guide](communication/style_guide.md).
+
+> NOTE: If your contribution is about using or configuring Sourcegraph (the product), it most likely belongs in the [Sourcegraph documentation](engineering/documentation.md).
```
Comment (nicksnyder) [inline handbook/usage.md]:
```suggestion
> NOTE: If your contribution is about using or configuring Sourcegraph (the product), it most likely belongs in the [product documentation](engineering/product_documentation.md).
```

File: handbook/engineering/documentation.md (hunk 2 above)
Comment (ryan-blunden) [reply handbook/engineering/documentation.md]: Yep, will rename.

**Pull request merged**